### PR TITLE
[Outlook] (signature) Convert sample to GitHub hosting

### PR DESCRIPTION
**Pull request by ElizabethSamuel-MSFT**
|        Q        |                    A                    |
| --------------- | --------------------------------------- |
| Bug fix?        | no - yes?  no                             |
| New feature?    | no - yes?   no                            |
| New sample?     | no - yes?    no                           |
| Related issues? | none |

## What's in this Pull Request?

Converting the outlook-set-signature sample from yo office project to GitHub hosting.

### CODE REVIEW

**Comment by aafvstam** [comment]
Hi Elizabeth,

I scanned the PR and think there are two attention areas:

1. You added the localhost manifest (and I agree, don't take away the localhost option, people will use this and replace the localhost info with their own server info) but I believe all the localhost/server instructions have been removed leaving only the GitHub use instructions.
2. The GitHub deployment is done in your own GitHub space. I think it would be better to deploy this in this GitHub repo as personal accounts tend to move, go away ... other

For the last item (2) I'd suggest changing https://elizabethsamuel-msft.github.io/ to https://officedev.github.io/ 

I checked and I even think it is already in there? Example:
https://officedev.github.io/PnP-OfficeAddins/Samples/outlook-set-signature/src/taskpane/HTML/editsignature.html

**Comment by davidchesnut** [comment]
Hi @aafvstam,

For item 2, we don't have a way to use GitHub pages for the hosting in a PR because you can't host from the PR branch (only main branch). So we use a personal fork to test, then update the manifest quickly right after we merge the PR. The change you suggest is made after the merge. If there's a way to avoid that it would be great to know!

Thanks!
David

**Comment by aafvstam** [comment]
@davidchesnut 
I think the files in https://officedev.github.io/ are independent of the PR, you can see for instance that https://officedev.github.io/PnP-OfficeAddins/Samples/outlook-set-signature/src/taskpane/HTML/editsignature.html already exists even though the PR is not merged yet.

Therefor the PR should be able to already use the https://officedev.github.io/... URL in code, don't you think?

@ElizabethSamuel-MSFT
Item 1 can be ignored, by cloning the PR and looking properly at the MD I see that you split it two ways just like I had in mind :-)

**Comment by ElizabethSamuel-MSFT** [comment]
@aafvstam Thanks for your feedback. I checked settings for this repo and Pages is set to "main". I think the files work from officedev.github.io because the sample already exists here - I'm only converting a sample but not adding a new one. I have no problem updating the links in this case because I didn't change any content in the various js and HTML files.

**Comment by aafvstam** [comment]
@ElizabethSamuel-MSFT Indeed, my updated comment crossed your answer. As the specific asset items haven't changed it should be possible to PR with the officedev.github.io URLs. 

Creating a PR with changed assets would run into the issues I described in the second 'update' ...

File: Samples/outlook-set-signature/README.md [hunk 1]
```diff
@@ -31,7 +31,9 @@ For documentation related to this sample, see [Configure your Outlook add-in for
 
 ## Applies to
 
-- Outlook on Windows, and on the web.
+- Outlook
+  - Windows
+  - web browser
 
 ## Prerequisites
```
Comment (davidchesnut) [inline Samples/outlook-set-signature/README.md:38]: The npm prereq is not needed with GitHub hosting. If someone wants to run localhost the steps are later, and should be self-evident npm would be needed for that particular scenario. I would just list a Microsoft 365 account (developer account) as the prereq.

**Review by davidchesnut — APPROVED** [review]
Just one comment. Otherwise looks good!

**Comment by aafvstam** [comment]
This is a good example of the testing chicken-egg issue mentioned above while testing the PR:

image

Here you see the breaking logo caused by 

image

This in fact is the 'old' page served by GitHub Pages (pre-merge PR).

If you check *your* code it is actually OK and should be resolved post-merge PR.

That said, I think you are OK to merge ... Deployment to GH Pages do require a post-merge check however 😉

**Pull request merged**